### PR TITLE
fix: add https-proxy-agent to fetch requests

### PR DESCRIPTION
**Pull request by malis42**
## What/Why/How?
Adding https-proxy-agent package to fix issues with requests when using CLI behind proxy. Create better mechanism for handling `env.REDOCLY_DOMAIN` usage in few modules. 

Now if you need to run the CLI tool behind a proxy, you can use the HTTP_PROXY and HTTPS_PROXY environment variables to configure the proxy settings.

`https-proxy-agent` package was used for `node-fetch` requests. Now we pass agent for every `fetchWithTimeout` request, as well as other regular `fetch` request.

## Reference
Resolves https://github.com/Redocly/redocly-cli/issues/1404
Fixes https://github.com/Redocly/redocly-cli/issues/529
Fixes https://github.com/Redocly/redocly-cli/issues/1146

## Testing
Fixed unit tests. Tested with few CLI commands: `push`, `lint` etc.

## Screenshots (optional)
n/a

## Has code been changed?
Yes

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by lornajane** [comment]
Cool! I think this relates to some open issue(s) though, could you link them from the PR description?

File: packages/cli/src/cms/api/api-client.ts [hunk 1]
```diff
@@ -70,6 +74,10 @@ class RemotesApiClient {
       }
     );
 
+    if (!response) {
+      throw new Error(`Failed to upsert: Request timed out.`);
```
Comment (IgorKarpiuk) [inline packages/cli/src/cms/api/api-client.ts:78]: if the request fails due to some other error, we still receive a message Request timeout

Comment (malis42) [reply packages/cli/src/cms/api/api-client.ts]: After discussion, we decided to change this message to more generic one.

**Review by tatomyr — COMMENTED** [review]
Please also test this in Workflows and add some docs for the new proxy feature.

File: packages/core/src/utils.ts [hunk 2]
```diff
@@ -274,3 +276,23 @@ export function nextTick() {
 function getUpdatedFieldName(updatedField: string, updatedObject?: string) {
   return `${typeof updatedObject !== 'undefined' ? `${updatedObject}.` : ''}${updatedField}`;
 }
+
+export function isRedoclyRegistryURL(link: string): boolean {
```
[the remark this answers is not in the document]
Comment (malis42) [reply packages/core/src/utils.ts]: Moved into the `domains.ts` file

File: packages/core/src/utils.ts [hunk 3]
```diff
@@ -5,9 +5,11 @@ import fetch from 'node-fetch';
 import * as pluralize from 'pluralize';
 import { parseYaml } from './js-yaml';
 import { UserContext } from './walk';
-import { HttpResolveConfig } from './config';
 import { env } from './env';
 import { logger, colorize } from './logger';
+import { DEFAULT_REGION, DOMAINS, getRedoclyDomain } from './domains';
+import { HttpResolveConfig } from './config';
+import { HttpsProxyAgent } from 'https-proxy-agent';
```
[the remark this answers is not in the document]
Comment (malis42) [reply packages/core/src/utils.ts:11]: Fixed!

File: .changeset/great-stingrays-unite.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": minor
+"@redocly/cli": minor
+---
+
+Fix an issue with certain requests failing when using CLI behind a proxy.
```
Comment (HCloward) [inline .changeset/great-stingrays-unite.md:6]:
```suggestion
Users can run the CLI tool behind a proxy by using `HTTP_PROXY` or `HTTPS_PROXY` environment variables to configure the proxy settings. 
```

File: packages/core/src/redocly/domains.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { Region } from '../config/types';
+
+let REDOCLY_DOMAIN = 'redocly.com';
```
[the remark this answers is not in the document]
Comment (malis42) [reply packages/core/src/redocly/domains.ts]: Fixed!

**Pull request merged**